src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -678,4 +678,25 @@ object messages {
            |$fixedVarInAlternative""".stripMargin
     }
   }
+
+  case class AuxConstructorNeedsNonImplicitParameter()(implicit ctx:Context) extends Message(24) {
+    val kind = "Syntax"
+    val msg = "auxiliary constructor needs non-implicit parameter list"
+    val code =
+      s"""class Person(val first: String, val last: String) {
+          |  def this() = this("John", "Doe")
+          |  def this(last: String) = this("John", last)
+          |}""".stripMargin
+
+    val explanation =
+      hl"""An auxiliary constructor requires a non-implicit parameter list. Auxiliary
+        |constructors are allowed to call either auxiliary or primary constructors,
+        |where the primary constructor is allowed implicit parameters. A common mistake
+        |is to forget the parenthesis on `this` (eg `def this() = ...`).
+        |
+        |Here's an example class with primary and auxiliary constructors:
+        |
+        |$code""".stripMargin
```

Comment:
Perhaps the reason as to why somebody would create an auxiliary constructor with an implicit parameter is because they get the error: `could not find implicit value for parameter X: Type`.

It might be worth mentioning that with auxiliary constructors calling a primary constructor that has an implicit argslist, the implicit value has to be explicitly specified at the callsite.